src/pages/ServiceList/ServiceListComponent.tsx

```diff
@@ -43,6 +49,7 @@ type ServiceListComponentState = {
   pagination: Pagination;
   currentSortField: SortField;
   isSortAscending: boolean;
+  rateInterval: string;
```

Comment:
rateInterval should be a typesafe `enum` since it is bounded.

I have not worried about these kinds of things yet, but we need to start tightening up on this now.

Reply:
It may be bounded in the component, but not in the model (the rest api accepts unbounded rate interval as long as it has the right format), so I don't know if it's really better to define an arbitrary enum type.

Reply:
I'm +1 with Joel's comment. Perhaps a better approach is to make an interface/class like this one:

```ts
interface TimeDuration {
    duration: number;
    timeUnit: TimeUnitEnum;
}
```

where `TimeUnitEnum` is a real enumeration with items for seconds, minutes, hours, etc. This way, we can query the rest API with any allowed rate interval rather than having a few predefined intervals.

Reply:
I think I'll fix this in a follow-up PR, so that we can better discuss how to deal with this.